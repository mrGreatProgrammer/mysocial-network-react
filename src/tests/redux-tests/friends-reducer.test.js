import friendsReducer, { setUsersAC } from "../../redux/friends-reducer";

let users = [
  {
    userId: 1,
    avatar: "avatar photo",
    name: "First",
    secondName: "Firstov",
    userSatus: "first status",
    followed: false,
  },
  {
    userId: 2,
    avatar: "avatar photo",
    name: "Second",
    secondName: "Secondov",
    userSatus: "second status",
    followed: false,
  },
  {
    userId: 3,
    avatar: "avatar photo",
    name: "Third",
    secondName: "Thirdov",
    userSatus: "third status",
    followed: false,
  },
  {
    userId: 4,
    avatar: "avatar photo",
    name: "Four",
    secondName: "Fourov",
    userSatus: "four status",
    followed: false,
  },
  {
    userId: 5,
    avatar: "avatar photo",
    name: "Five",
    secondName: "Ivanov",
    userSatus: "Five status",
    followed: false,
  },
  {
    userId: 6,
    avatar: "avatar photo",
    name: "Ivan",
    secondName: "Secondov",
    userSatus: "second status",
    followed: false,
  },
  {
    userId: 7,
    avatar: "avatar photo",
    name: "Seven",
    secondName: "Sevenov",
    userSatus: "Seven status",
    followed: false,
  },
];

let state = {
  friends: [],
  currentPage: 1,
  pageSize: 3,
  totalUsersCount: 0,
  isFetching: false,
};

describe("--------- friends reducer set users ----------", () => {
  it("SET_USERS", () => {
    let action = setUsersAC(users);
    let newState = friendsReducer(state, action);
    expect(newState.friends.length).toBe(7)
    expect(newState.friends).toMatchSnapshot()
  });
});
