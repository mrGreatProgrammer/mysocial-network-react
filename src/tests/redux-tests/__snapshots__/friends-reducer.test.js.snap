// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--------- friends reducer set users ---------- SET_USERS 1`] = `
Array [
  Object {
    "avatar": "avatar photo",
    "followed": false,
    "name": "First",
    "secondName": undefined,
    "userId": undefined,
    "userSatus": undefined,
  },
  Object {
    "avatar": "avatar photo",
    "followed": false,
    "name": "Second",
    "secondName": undefined,
    "userId": undefined,
    "userSatus": undefined,
  },
  Object {
    "avatar": "avatar photo",
    "followed": false,
    "name": "Third",
    "secondName": undefined,
    "userId": undefined,
    "userSatus": undefined,
  },
  Object {
    "avatar": "avatar photo",
    "followed": false,
    "name": "Four",
    "secondName": undefined,
    "userId": undefined,
    "userSatus": undefined,
  },
  Object {
    "avatar": "avatar photo",
    "followed": false,
    "name": "Five",
    "secondName": undefined,
    "userId": undefined,
    "userSatus": undefined,
  },
  Object {
    "avatar": "avatar photo",
    "followed": false,
    "name": "Ivan",
    "secondName": undefined,
    "userId": undefined,
    "userSatus": undefined,
  },
  Object {
    "avatar": "avatar photo",
    "followed": false,
    "name": "Seven",
    "secondName": undefined,
    "userId": undefined,
    "userSatus": undefined,
  },
]
`;
